research/2022-11-02-building-privacy-protecting-infrastructure.md

```diff
@@ -0,0 +1,489 @@
+sacrifices in how they function, from the way domains work, to how websites are
+hosted and the reliance on centralized services for communication. We see this
+time and time again, where centralized, single points of failure systems works
+for a while, but then eventually fails.
+
+In many cases an individual user might not care enough though, and for platforms
+the lure to take shortcuts is strong. That is why it is important to be
+principled, but also pragmatic in terms of the trade-offs that you allow on top.
+We'll touch more on this in the design goals around modularity that Waku has.
+
+### ZK for Privacy-protecting infrastructure
+
+ZKPs are a wonderful new tool. Just like smart contracts enables programmable
+money, ZKPs allow us to express fundamentally new things. In line with the great
+tradition of trust-minimization, we can prove statement while revealing the
+absolute minimum information necessary. This fits the definition of privacy, the
+power to selectively reveal yourself, perfectly. I'm sure I don't need to tell
+anyone in this room but this is truly revolutionary. The technology is advancing
+extremely fast and often it is our imagination that is the limit.
+
+![Zero knowledge](/img/building_private_infra_zk.png)
+
+### Waku
+
+What is Waku? It is a set of modular protocols for p2p communication. It has a
+focus on privacy, security and being able to run anywhere. It is the spiritual
+success to Whisper.
+
+By modular we mean that you can pick and choose protocols and how you use them
+depending on constraints and trade-offs. For example, bandwidth usage vs
+privacy.
+
+It is designed to work in resource restricted environments, such as mobile
+phones and in web browsers. It is important that infrastructure meets users
+where they are and supports their real-world use cases. Just like you don't need
+your own army and a castle to have your own private bathroom, you shouldn't need
+to have a powerful always-on node to get reasonable privacy and
+censorship-resistance. We might call this self-sovereignty.
+
+### Waku - adaptive nodes
+
+One way of looking at Waku is as an open service network. There are nodes with
+varying degrees of capabilities and requirements. For example when it comes to
+bandwidth usage, storage, uptime, privacy requirements, latency requirements,
+and connectivity restrictions.
+
+We have a concept of adaptive nodes that can run a variety of protocols. A node
+operator can choose which protocols they want to run. Naturally, there'll be
+some nodes that do more consumption and other nodes that do more provisioning.
+This gives rise to the idea of a service network, where services are provided
+for and consumed.
+
+![Adaptive Nodes](/img/building_private_infra_adaptive.png)
+
+### Waku - protocol interactions 
+
+There are many protocols that interact. Waku Relay protocol is based libp2p
+GossipSub for p2p messaging. We have filter for bandwidth-restricted nodes to
```

Comment:
I'd put links to the rfc of the protocols mentioned in here like filter, lightpush, etc.